### ISSUE TITLE
Function with 'yield from'/'yield' has false positives about return statements

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#970]
https://getpsalm.org/r/f36e0fb70f

```php
<?php

function example() : Generator {
    yield from [2];
    return null;
}
foreach (example() as $x) {
    echo $x;
}

function example2() : Generator {
    yield from [2];
    return null;
}
```

Observed: ERROR: NullableReturnStatement - 5:5 - The declared return type 'Generator' for example is not nullable, but the function returns 'null'
Expected: Should not emit NullableReturnStatement about any return statement in a function with `yield` or `yield from`

See https://secure.php.net/manual/en/generator.getreturn.php#refsect1-generator.getreturn-examples
